.github/workflows/github_pages.yml

```diff
@@ -33,18 +43,31 @@ jobs:
       - name: Set up Python
         uses: actions/setup-python@v4
         with:
-          python-version: "3.11"
+          python-version: ${{ inputs.python }}
+      - name: Checkout theme
+        if: ${{ inputs.theme }}
+        run: git clone '${{ inputs.theme }}' .theme
       - name: Configure GitHub Pages
         id: pages
         uses: actions/configure-pages@v3
       - name: Install requirements
         run: pip install ${{ inputs.requirements }}
       - name: Build Pelican site
+        shell: python
         run: |
-          pelican \
-            --settings "${{ inputs.settings }}" \
-            --extra-settings SITEURL='"${{ steps.pages.outputs.base_url }}"' \
-            --output "${{ inputs.output-path }}"
+          import subprocess
+
+          cmd = "pelican"
+          cmd += " --settings ${{ inputs.settings }}"
+          cmd += " --extra-settings"
+          cmd += """ SITEURL='"${{ steps.pages.outputs.base_url }}"'"""
+          cmd += """ FEED_DOMAIN='"${{ steps.pages.outputs.origin }}"'"""
```

Comment:
Adding the `FEED_DOMAIN` setting, along with the `SITEURL` setting that was already here.

GitHub Actions knows the URL of the GitHub Pages site that is being deployed to, so there's no need for the user to supply the `SITEURL` and `FEED_DOMAIN` settings, the workflow can inject them automatically.

`${{ steps.pages.outputs.base_url }}` is the base URL that the site is being deployed to, appropriate for the `SITEURL` setting. For example `"https://octocat.github.io"` (for a user or organization site) or `"https://octocat.github.io/my-repo"` for a project site. If the user has set up a custom domain for GitHub Pages it'll use that, e.g. `"https://www.example.com"` or `"https://www.example.com/my-repo"`.

`${{ steps.pages.outputs.origin }}` is the origin part of the site URL, for example `"https://octocat.github.io"` or `"https://www.example.com/my-repo"` (no `/my-repo` at the end for project sites). I believe this is correct for `FEED_DOMAIN`?

(These variables come from the output's of GitHub's `configure-pages` action, they're documented here: <https://github.com/actions/configure-pages/blob/e1bedb377708461b49c882870df192bdec04b996/action.yml#L22-L30>.)

Reply:
I added the `FEED_DOMAIN` setting for the use case in which the domain used for the web site (e.g., `example.com`) is different than the domain used for serving the feed(s) (e.g., `feeds.example.com`). If, on the other hand, both the web site and the feeds are served from the same domain, then the appropriate setting should be: `FEED_DOMAIN = SITEURL`

Does that help clarify?

Reply:
Got it, thanks.

Ok, I've changed it to set `FEED_DOMAIN` to the same URL of the GitHub Pages site as it sets `SITEURL` to. I've tested this on two of my own Pelican sites, one a user site and one a project site (project sites are deployed to sub-folders of the user's domain) and it seems to work correctly.

I also added an optional `feed_domain` workflow input so that users can override the default `FEED_DOMAIN` in case they're somehow deploying their feeds to a different domain.

Also added a `siteurl` input to override the default `SITEURL`. I'm not sure whether there's a use-case for setting `SITEURL` manually with this workflow but I put it in just for good measure.

I also added a note to the docs about GitHub Pages generating `http://` URLs for `https://` sites and how to fix it.